pallets/dca/src/lib.rs

```diff
@@ -0,0 +1,1073 @@
+
+		match &schedule.order {
+			Order::Sell {
+				asset_in,
+				asset_out,
+				amount_in,
+				min_limit,
+				slippage,
+				route,
+			} => {
+				let remaining_amount_to_use =
+					RemainingAmounts::<T>::get(schedule_id).defensive_ok_or(Error::<T>::InvalidState)?;
+				let amount_to_sell = min(remaining_amount_to_use, *amount_in);
+
+				Self::unallocate_amount(schedule_id, schedule, amount_to_sell)?;
+
+				let (estimated_amount_out, slippage_amount) =
+					Self::calculate_estimated_and_slippage_amounts(*asset_out, *asset_in, amount_to_sell, *slippage)?;
+				let min_limit_with_slippage = estimated_amount_out
+					.checked_sub(slippage_amount)
+					.ok_or(ArithmeticError::Overflow)?;
+				let min_limit = max(*min_limit, min_limit_with_slippage);
+
+				let route = Self::convert_to_vec(route);
+				let trade_amounts =
+					pallet_route_executor::Pallet::<T>::calculate_sell_trade_amounts(&route, amount_to_sell.into())?;
+				let last_trade = trade_amounts.last().defensive_ok_or(Error::<T>::InvalidState)?;
+				let amount_out = last_trade.amount_out;
+
+				if amount_out < min_limit.into() {
+					return Err(Error::<T>::TradeLimitReached.into());
+				}
+
+				pallet_route_executor::Pallet::<T>::sell(
+					origin,
+					(*asset_in).into(),
+					(*asset_out).into(),
+					(amount_to_sell).into(),
+					min_limit.into(),
+					route,
+				)
+			}
+			Order::Buy {
+				asset_in,
+				asset_out,
+				amount_out,
+				slippage,
+				max_limit,
+				route,
+			} => {
+				let amount_in = Self::get_amount_in_for_buy(amount_out, route)?;
+
+				Self::unallocate_amount(schedule_id, schedule, amount_in.into())?;
+
+				let (estimated_amount_in, slippage_amount) =
+					Self::calculate_estimated_and_slippage_amounts(*asset_in, *asset_out, *amount_out, *slippage)?;
+				let max_limit_with_slippage = estimated_amount_in
+					.checked_add(slippage_amount)
+					.ok_or(ArithmeticError::Overflow)?;
+
+				let max_limit = min(*max_limit, max_limit_with_slippage);
+				if amount_in > max_limit.into() {
+					return Err(Error::<T>::TradeLimitReached.into());
+				}
+
+				pallet_route_executor::Pallet::<T>::buy(
+					origin,
+					(*asset_in).into(),
+					(*asset_out).into(),
+					(*amount_out).into(),
+					max_limit.into(),
+					Self::convert_to_vec(route),
+				)
+			}
+		}
+	}
+
+	fn replan_or_complete(
+		schedule_id: ScheduleId,
+		schedule: &Schedule<T::AccountId, T::Asset, T::BlockNumber>,
+		current_blocknumber: T::BlockNumber,
+	) -> DispatchResult {
+		Self::deposit_event(Event::TradeExecuted {
+			id: schedule_id,
+			who: schedule.owner.clone(),
+		});
+
+		Self::reset_retries(schedule_id)?;
+
+		let remaining_amount_to_use: T::Balance = RemainingAmounts::<T>::get(schedule_id)
+			.defensive_ok_or(Error::<T>::InvalidState)?
+			.into();
+		let transaction_fee = Self::get_transaction_fee(&schedule.order)?;
+
+		if remaining_amount_to_use < transaction_fee.into() {
```

Comment:
i believe here should be <= ? 

You can complete also when amount is just transtaction fee.